lib/bitcoinerlab.js

```diff
@@ -0,0 +1,144 @@
+  return array.length === 33
+}
+
+module.exports = {
+  contextRandomize: function () {
+    return 0
+  },
+  privateKeyVerify: function (privateKey) {
+    return secp256k1.isPrivate(privateKey) ? 0 : 1
+  },
+  privateKeyNegate: function (privateKey) {
+    try {
+      privateKey.set(secp256k1.privateNegate(privateKey))
+    } catch (e) {
+      loadLegacy('privateKeyNegate', e.toString())
+      elliptic.privateKeyNegate(privateKey)
+    }
+    return 0
+  },
+  privateKeyTweakAdd: function (privateKey, tweak) {
+    try {
+      privateKey.set(secp256k1.privateAdd(privateKey, tweak))
+    } catch (e) {
+      return 1
+    }
+    return 0
+  },
+  privateKeyTweakMul: function (privateKey, tweak) {
+    loadLegacy('privateKeyTweakMul')
+    return elliptic.privateKeyTweakMul(privateKey, tweak)
+  },
+  publicKeyVerify: function (publicKey) {
+    return secp256k1.isPoint(publicKey) ? 0 : 1
+  },
+  publicKeyCreate: function (output, seckey) {
+    try {
+      output.set(secp256k1.pointFromScalar(seckey, isCompressedPublicKey(output)))
+      return 0
+    } catch (e) {
+      if (e.toString().includes('Expected Private')) return 1
+      return 2
+    }
+  },
+  publicKeyConvert: function (output, pubkey) {
+    try {
+      output.set(secp256k1.pointCompress(pubkey, isCompressedPublicKey(output)))
+      return 0
+    } catch (e) {
+      if (e.toString().includes('Expected Point')) return 1
+      return 2
+    }
+  },
+  publicKeyNegate: function (output, pubkey) {
+    loadLegacy('publicKeyNegate')
+    return elliptic.publicKeyNegate(output, pubkey)
+  },
+  publicKeyCombine: function (output, pubkeys) {
+    loadLegacy('publicKeyCombine')
+    return elliptic.publicKeyCombine(output, pubkeys)
+  },
+  publicKeyTweakAdd: function (output, pubkey, tweak) {
+    try {
+      output.set(secp256k1.pointAddScalar(pubkey, tweak, isCompressedPublicKey(output)))
+      return 0
+    } catch (e) {
+      if (e.toString().includes('Expected Point')) return 1
+      return 2
+    }
+  },
+  publicKeyTweakMul: function (output, pubkey, tweak) {
+    try {
+      output.set(secp256k1.pointMultiply(pubkey, tweak, isCompressedPublicKey(output)))
+      return 0
+    } catch (e) {
+      if (e.toString().includes('Expected Point')) return 1
+      return 2
+    }
+  },
+  signatureNormalize: function (sig) {
+    loadLegacy('signatureNormalize')
+    return elliptic.signatureNormalize(sig)
+  },
+  signatureExport: function (obj, sig) {
+    loadLegacy('signatureExport')
+    return elliptic.signatureExport(obj, sig)
+  },
+  signatureImport: function (output, sig) {
+    loadLegacy('signatureImport')
+    return elliptic.signatureImport(output, sig)
+  },
+  ecdsaSign: function (obj, msg32, seckey, data, noncefn) {
+    if (noncefn) {
+      loadLegacy('ecdsaSign', 'fallback to legacy')
+      return elliptic.ecdsaSign(obj, msg32, seckey, data, noncefn)
+    }
+    try {
+      obj.signature.set(secp256k1.sign(msg32, seckey, data))
```

Comment:
for simplicity, can we completely default to the old implementation?

Reply:
`recid` is rarely required, I prefer to leave it as is

Reply:
@alexandrius is ecdsaSign very slow without BigInt? I think what @kewde meant was we don't have to use the BigInt implementation here at all

Reply:
Gotta check that

Reply:
We can have our cake and eat it too. `@noble/secp256k1` returns the recovery bit if you pass the `{ recovered: true }` option to the `sign` function. 

https://www.npmjs.com/package/@noble/secp256k1/v/1.7.1#signmsghash-privatekey

It doesn't support custom nonce functions, but that's probably for the best. In fact, i would be tempted to actually knee-cap the `noncefn` parameter, so that it has zero effect on the signature. There's no good reason (outside maliciousness) for a caller to use that when RFC6979 deterministic nonce derivation is implemented.